support-lib/cpp/Future.hpp

```diff
@@ -364,34 +366,63 @@ class Future {
         return true;
     }
 
+    template<typename ConcretePromise>
     struct PromiseTypeBase {
         Promise<T> _promise;
+        std::optional<djinni::expected<T, std::exception_ptr>> _result{};
+
+        struct FinalAwaiter {
+            constexpr bool await_ready() const noexcept {
+                return false;
+            }
+            bool await_suspend(std::coroutine_handle<ConcretePromise> finished) const noexcept {
```

Comment:
Let's use the alias `detail::CoroutineHandle` for now

Reply:
Done 👍